src/bibliography_node.jl

```diff
@@ -0,0 +1,177 @@
+"""Representation of a reference within a [`BibliographyNode`}(@ref).
+
+# Properties
+
+* `anchor_key`: the anchor key for the link target as a string (generally the
+  BibTeX key), or `nothing` if the item is in a non-canonical bibliography
+  block.
+* `label`: `MarkdownAST.Node` for the label of the entry. Usually a simple
+  `Text` node, as labels in the default styles do not have inline formatting.
+    or `nothing` if the style does not use labels
+  the rendered bibliography.
+* `reference`: `MarkdownAST.Node` for the paragraph for the fully rendered
+  reference.
+"""
+struct BibliographyItem
+    anchor_key::Union{Nothing,String}
+    label::Union{Nothing,MarkdownAST.Node{Nothing}}
+    reference::MarkdownAST.Node{Nothing}
+end
+
+
+"""Node in `MarkdownAST` corresponding to a `@bibliography` block.
+
+# Properties
+
+* `list_style`: One of `:dl`, `:ul`, `:ol`, cf. [`bib_html_list_style`](@ref)
+* `canonical`: Whether or not the references in the `@bibliography` block are
+  link targets.
+* `items`: A list of [`BibliographyItem`](@ref) objects, one for each reference
+in the block
+"""
+struct BibliographyNode <: Documenter.AbstractDocumenterBlock
+    list_style::Symbol  # one of :dl, :ul, :ol
+    canonical::Bool
+    items::Vector{BibliographyItem}
```

Comment:
```suggestion
    list_style::Symbol  # one of :dl, :ul, :ol
    canonical::Bool
    items::Vector{BibliographyItem}
    function BibliographyNode(style, canon, items)
        style in (:dl, :ul, :ol) || throw(ArgumentError("Only :dl, :ul or :ol is allowed as a list style!"))
        new(style, canon, items)
    end
```

Better to check these assumptions than throw later on, if this is assumed throughout the code :)

Reply:
Fair enough, although this is already checked earlier in the code. So the only way for a user to trigger that error would be to instantiate a `BibliographyNode` manually.